OneCommunityGlobal/HGNRest
# Fix X hours in X weeks streak badges

# Description
![picturetwo](https://github.com/OneCommunityGlobal/HGNRest/assets/91035340/8cd31bba-dd39-4b02-a122-f836b2817fb6)

## Related PRS (if any):
To test this backend PR you need to check the latest frontend

## Main changes explained:
it was included a simple logic in checkXHoursXWeeks function
…

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. you need to add to run awardNewBadges every time 
5. you need to take off awardNewBadges function inside condicional
6. you need to enter in the mongodb compass
7. you need to check userProfile folder
8. you need to add the hours you would like to get the badge streak in savedTangibleHours array
9. you need to monitor if in BadgeCollections appear only one badge streak

see the tutorials how to test below:

https://github.com/OneCommunityGlobal/HGNRest/assets/91035340/5faaf837-188c-40c3-b12c-92b8bc32e720


https://github.com/OneCommunityGlobal/HGNRest/assets/91035340/6c8ecfc2-46c2-4f01-879b-4fed93ff3dc6


https://github.com/OneCommunityGlobal/HGNRest/assets/91035340/9d00e7cc-b787-4c67-8f87-211ea7387ccf





## Screenshots or videos of changes:

## Note:
Include the information the reviewers need to know.
1. if the badge didn't appear in the badgeCollection, probably the badge don't exist
2. if you have some difficults to test the back-end in nodemon, please watch this video: https://www.youtube.com/watch?v=GYR6GSAVC_k or this video https://www.dropbox.com/scl/fi/223fqqbzultw6chbl9a5p/Backend-testing-guide.mkv?rlkey=ygsr5w91lf76yxdha9n83favs&dl=0

Some questions please send me a dm in slack
